benchmarks/transformer/score_mod.py

```diff
@@ -6,25 +6,29 @@
 
 import numpy as np
 import torch
-import torch.utils.benchmark as benchmark
+import torch.nn.functional as F
 from tabulate import tabulate
-from torch.nn.attention._templated_attention import _compose, _templated_attention
+from torch.nn.attention._templated_attention import _templated_attention
 from tqdm import tqdm
 
 torch._dynamo.config.automatic_dynamic_shapes = False
 # Needed since changing args to function causes recompiles
 torch._dynamo.config.cache_size_limit = 1000
 
 
+from triton.testing import do_bench
+
+
 def benchmark_torch_function_in_microseconds(func: Callable, *args, **kwargs) -> float:
     # warmup
     for _ in range(5):
         func(*args, **kwargs)
-    t0 = benchmark.Timer(
-        stmt="func(*args, **kwargs)",
-        globals={"args": args, "kwargs": kwargs, "func": func},
-    )
-    return t0.adaptive_autorange(min_run_time=0.1).median * 1e6
+    return do_bench(lambda: func(*args, **kwargs)) * 1e3
```

Comment:
This hides kernel launch time overhead but one criticism I have a seen is the increased time to launch, hence why I preferred this.